tabular/src/autogluon/tabular/models/tabular_nn/torch/tabular_torch_dataset.py

```diff
@@ -118,6 +121,14 @@ def __init__(self, processed_array, feature_arraycol_map, feature_type_map, prob
                     labels = labels.astype('long')
                 self.data_list.append(labels.reshape(-1, 1))
 
+        # base prediction data
+        if base_preds is not None:
+            base_preds = np.array(base_preds)
+            self.data_desc.append("base_pred")
+            self.base_pred_index = len(self.data_list)
```

Comment:
I feel like this aggregator-specific code is confusing to have in the general `TabularTorchDataset` which is intended for standard pytorch neural nets trained on the original data, not aggregator networks.  

At the very least it would be good to have a comment clarifying what that this code is specific for aggregator networks.  Even better could be if aggregator networks use a different `aggregator_torch_dataset` object that is inherits from `TabularTorchDataset`.